packages/iov-keycontrol/src/keyring.spec.ts

```diff
@@ -15,204 +15,226 @@ async function makeRandomEd25519Keypair(): Promise<Ed25519Keypair> {
-    const i4c = await wallet4.createIdentity(defaultChain, await makeRandomEd25519Keypair());
-    const i4d = await wallet4.createIdentity(defaultChain, await makeRandomEd25519Keypair());
-
-    const keyring = new Keyring();
-    keyring.add(wallet1);
-    keyring.add(wallet2);
-    keyring.add(wallet3);
-    keyring.add(wallet4);
-
-    const serialized = keyring.serialize();
-
-    const restored = new Keyring(serialized);
-    expect(restored).toBeTruthy();
-
-    // compare wallets
-
-    expect(restored.getWallets().length).toEqual(4);
-    expect(keyring.getWallets()[0]).toEqual(jasmine.any(Ed25519HdWallet));
-    expect(keyring.getWallets()[0].getIdentities().length).toEqual(1);
-    expect(keyring.getWallets()[1]).toEqual(jasmine.any(Ed25519Wallet));
-    expect(keyring.getWallets()[1].getIdentities().length).toEqual(2);
-    expect(keyring.getWallets()[2]).toEqual(jasmine.any(Ed25519HdWallet));
-    expect(keyring.getWallets()[2].getIdentities().length).toEqual(3);
-    expect(keyring.getWallets()[3]).toEqual(jasmine.any(Ed25519Wallet));
-    expect(keyring.getWallets()[3].getIdentities().length).toEqual(4);
-
-    // compare wallet content (via PublicIdentity equality)
-
-    expect(keyring.getWallets()[0].getIdentities()[0]).toEqual(i1a);
-
-    expect(keyring.getWallets()[1].getIdentities()[0]).toEqual(i2a);
-    expect(keyring.getWallets()[1].getIdentities()[1]).toEqual(i2b);
-
-    expect(keyring.getWallets()[2].getIdentities()[0]).toEqual(i3a);
-    expect(keyring.getWallets()[2].getIdentities()[1]).toEqual(i3b);
-    expect(keyring.getWallets()[2].getIdentities()[2]).toEqual(i3c);
-
-    expect(keyring.getWallets()[3].getIdentities()[0]).toEqual(i4a);
-    expect(keyring.getWallets()[3].getIdentities()[1]).toEqual(i4b);
-    expect(keyring.getWallets()[3].getIdentities()[2]).toEqual(i4c);
-    expect(keyring.getWallets()[3].getIdentities()[3]).toEqual(i4d);
-  });
+    it("can serialize many wallets", () => {
+      const keyring = new Keyring();
+      const wallet1 = Ed25519HdWallet.fromEntropy(
+        fromHex("c7f74844892fd7b707e74fc9b6c8ef917c13ddbb380cadbc"),
+      );
+      const wallet2 = Ed25519HdWallet.fromEntropy(
+        fromHex("2a7e3f902279af82138f14f871badf8d92b33713eb6c7193"),
+      );
+      const wallet3 = Ed25519HdWallet.fromEntropy(
+        fromHex("602c79484cf098bd4445ad45b5e6557d83ec743cebddb4cd"),
+      );
+      const wallet4 = Ed25519HdWallet.fromEntropy(
+        fromHex("1124ef7ab681387eba8fdd93a0a88ec2f3326f2a6e5e967d"),
+      );
+      const wallet5 = Ed25519HdWallet.fromEntropy(
+        fromHex("cd0e346ae5c714a1514562cb8ad5d4b5a2443dbbc5dd2b5b"),
+      );
+      const wallet6 = Ed25519HdWallet.fromEntropy(
+        fromHex("e38dd5c066406668b51be00e8ad0276ed9dec967d95c2248"),
+      );
+      keyring.add(wallet1);
+      keyring.add(wallet2);
+      keyring.add(wallet3);
+      keyring.add(wallet4);
+      keyring.add(wallet5);
+      keyring.add(wallet6);
```

Comment:
Consider a `.map` or `.forEach` function for an array of entropy strings.

Reply:
Thanks, done in add96e16669edce374f503f5a60bbea8b86a0cb0

Reply:
Sorry, no, it was 4f58cf0b853a2a18c5502124ae0ff9352647e3a4